build_scripts/build_addon.sh

```diff
@@ -8,34 +8,56 @@
 #  REQUIREMENTS: ---
 #          BUGS: ---
 #         NOTES: ---
-#        AUTHOR:  (),
+#        AUTHOR:  (), Lilach Maliniak
 #  ORGANIZATION: Kaltura, inc.
-#       CREATED:
+#       CREATED: June 25 2017
 #      REVISION:  ---
 #===============================================================================
-
-if [ "$#" -lt 1 ]; then
-	echo "usage build_addon <product path> [Release/Debug]"
+set +e
+if [ "$#" -lt 2 ]; then
+	echo "usage build_addon <ffmpeg path> <product path> [Release/Debug]"
 	exit 1
 fi
 
-PRODUCT_ROOT_PATH=$1
+FFMPEG_PATH=$1
+PRODUCT_ROOT_PATH=$2
 BUILD_CONF=Release
-ADDON_PATH=${PRODUCT_ROOT_PATH}/node_addons/FormatConverter
+ADDON_BUILD_PATH=${PRODUCT_ROOT_PATH}/node_addons/FormatConverter/build
+FFMPEG_SYMLINK=${ADDON_BUILD_PATH}/FFmpeg
 FORMAT_CONVERTER_BIN=FormatConverter.so
 OS=`uname`
 RES=0
 
 echo "OS=$OS"
 
-[ "$2" = "Debug" ] && BUILD_CONF=Debug
+[ "$3" = "Debug" ] && BUILD_CONF=Debug
 
+echo "current path `pwd`"
+echo "PRODUCT_ROOT=${PRODUCT_ROOT}"
+echo "FFMPEG_PATH=${FFMPEG_PATH}"
 echo "BUILD_CONF=${BUILD_CONF}"
-echo "ADDON_PATH=${ADDON_PATH}"
 
 mkdir -p ${PRODUCT_ROOT_PATH}/bin
+mkdir -p ${ADDON_BUILD_PATH}
+
+# note: if the second argument already exists and is a directory,
+# ln will create a symlink to the target inside that directory.
 
-pushd ${ADDON_PATH}
+if [  -L ${FFMPEG_SYMLINK} ]; then
+	echo "unlink ${FFMPEG_SYMLINK}"
+	unlink ${FFMPEG_SYMLINK}
+fi
+
+if [ ! -r ${FFMPEG_SYMLINK} ]; then
+	echo "ln -s ${FFMPEG_PATH} ${FFMPEG_SYMLINK}"
+	ln -s ${FFMPEG_PATH} ${FFMPEG_SYMLINK}
+else
+	echo "failed to unlink ${FFMPEG_PATH}"
+	exit 1
```

Comment:
Always best to use different RCs per case. Since you're already exiting with 1 in the event you didn't get the required args, exit with 2 here.